### PR TITLE
DIGEQ-55 AWS terraform configuration

### PR DESCRIPTION
**Pull request by warren-methods**
**What**
Amazon EC2 Container Service (ECS) configuration for the eq-survey runner. This deploys 3 docker containers to EC2:
1] A python based container hosting gunicorn and our eq-survey application
2] An nginx container configured as a reverse proxy to our eq-survey application
3] A cassandra instance based on the spotify image

**How to test**
1] Check out this branch
2] Cd into the aws directory
3] Ensure that you have signed up for a aws account and have generated access, secret keys and a key pair.
4] Copy `terraform.tfvars.example` to `terraform.tfvars` and replace your aws details in the file.
5] Run `terraform plan` to check the output of terraform.
6] Run `terraform apply`  to create your infrastructure environment.
7] The output for survey runner is the DNS name of the load balancer. You can use this to access the eq-survey-runner service.

If you see this error, then it is most likely a timing issue with Role/Policy propagation. Run terraform apply again and it should then succeed

`Error applying plan:`

`1 error(s) occurred:`

`* aws_ecs_service.eq-survey-runner: InvalidParameterException: Unable to assume role and validate the listeners configured on your load balancer.  Please verify the role being passed has the proper permissions.`
`	status code: 400, request id: []`

`Terraform does not automatically rollback in the face of errors.`
`Instead, your Terraform state file has been partially updated with `
`any resources that successfully completed. Please address the error`
`above and apply again to incrementally change your infrastructure.`

**Who can review**
Anyone but @warren-methods

### CODE REVIEW

File: eq_paas_deploy.tf [hunk 1]
```diff
@@ -11,7 +11,7 @@ resource "heroku_app" "eq_author" {
 
     config_vars {
         # Kludge to get around cyclic issue. BEWARE. @TOFIX
```
Comment (dhilton) [inline eq_paas_deploy.tf]: This is fixed actually by this PR so we can remove the comment :+1:

**Comment by dhilton** [comment]
I think we need to namespace all the created parts with the environment - I tripped up on recreating an ecsRole that you had previously created. The name of the ELB also needs namespacing similarly. Other than that all looks good.

**Comment by warren-methods** [comment]
Yep that makes sense, essentially everything needs to be unique to the user. I'll sort that out now.

**Comment by dhilton** [comment]
This LGTM :+1: It all works apart from the docker hub issue which seems like a service issue.

Mergin'

**Pull request merged**